src/main/java/pl/project13/maven/git/GitDataProvider.java

```diff
@@ -112,8 +117,11 @@ public void loadGitData(@NotNull Properties properties) throws IOException, Mojo
       // git.commit.id.describe
       maybePutGitDescribe(properties);
       // git.commit.id
-      put(properties, GitCommitIdMojo.COMMIT_ID, getCommitId());
-      // git.commit.id.abbrev      
+      switch (commitIdGenerationMode) {
+        case FULL: put(properties, GitCommitIdMojo.COMMIT_ID_FULL, getCommitId());
+        default: put(properties, GitCommitIdMojo.COMMIT_ID_FLAT, getCommitId());
```

Comment:
I'd prefer `case FULL / case FLAT` in case we add a new mode it's easier to spot the need to handle here then here